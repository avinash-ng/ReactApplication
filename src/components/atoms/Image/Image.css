img {
    width: 450px;
    height: 450px;
    float: left;
    margin-left: 20px;
    margin-top: 0px;
    padding-top: 60px;
    /* margin-bottom: 30px; */
}