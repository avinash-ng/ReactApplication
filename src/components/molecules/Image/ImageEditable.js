import React from 'react'
import ImageTag from '../../atoms/ImageTag/ImageTag'
import Button from "../../atoms/Button/Button"
const ImageEditable = props => {
    return (
        <div className="tag-div" contentEditable="true" suppressContentEditableWarning={true}>
            {/* <h3 className = "h3-image-tag" style={{ width: '450px' }}  onClick={() => changeHeading(each, index)} >{each.name}</h3> */}
            <ImageTag className={props.className} onClicked={props.headingOnClicked} object={props.object}></ImageTag>
            {props.object.selected ? <Button buttonName={props.buttonName} onClicked={props.buttonOnClicked}></Button> : ""}

        </div>
    )
}
export default ImageEditable