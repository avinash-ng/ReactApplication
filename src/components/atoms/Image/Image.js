import React from 'react'
import PropTypes from "prop-types"
import "./Image.css"
const Image = (props) => {
    return (
        <img data-testid="image" src={props.imageName} alt={props.altText} />
    )
}
Image.propTypes = {
    imageName: PropTypes.string.isRequired,
    altText: PropTypes.string.isRequired
} 

export default Image