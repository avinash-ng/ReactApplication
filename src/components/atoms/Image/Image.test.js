import Image from './Image'
import expect from 'expect'
import sinon from 'sinon';
import { shallow, configure, mount } from 'enzyme'
import React from 'react'
import Adapter from 'enzyme-adapter-react-16';
// import renderer from 'react-test-renderer'
import { render } from '@testing-library/react';

configure({ adapter: new Adapter() });
describe('<Image /> ', () => {
    
    
    // it('renders alt text', () => {
    //     const RenderComponent = mount(
    //         <Image imageName="max.png" altText="alternate text..."></Image>
    //     );
    //     // const image = getByTestId("image")
    //     console.log(RenderComponent.find('img'))
    //     expect(RenderComponent)
    // })

    
    it('recieved props correctly', () => {
        const alt = "Error in loading..."
        const srcImage = "photo.png"
        const RenderComponent = shallow(<Image imageName={srcImage} altText={alt}></Image>)
        expect(RenderComponent.find('img').prop('alt')).toEqual(alt)

    })


    it('Matches snapshot', () => {
        const imageName = "max.png"
        const altText="Error loading max..."
        const RenderComponent = shallow(<Image imageName={imageName} altText = {altText}></Image>)
        expect(RenderComponent).toMatchSnapshot()
    
    })

})