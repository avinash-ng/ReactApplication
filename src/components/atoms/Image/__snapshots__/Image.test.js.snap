// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Image />  Matches snapshot 1`] = `ShallowWrapper {}`;
